packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -124,6 +156,12 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
       }
     }
 
+    //When the requestType is delete, delete the log group if the removal policy is delete
+    if (event.RequestType === 'Delete' && logDeletionPolicy === 'destroy') {
```

Comment:
I would test this directly against the enum, not the string it's set to. `RemovalPolicy.DESTROY`